dna/zomes/courses/code/src/course.rs

```diff
@@ -93,47 +137,84 @@ pub fn course_entry_def() -> ValidatingEntryType {
             }
         },
         links: [
-          from!( // to query all the courses of a user(all courses that a user is the teacher or owner of)
-              "%agent_id",
-              link_type: "teacher->courses",
-              validation_package: || {
-                  hdk::ValidationPackageDefinition::Entry
-              }              ,
-              validation: | _validation_data: hdk::LinkValidationData | {
-                // TODO: Homework. Implement validation rules if required.
-                Ok(())
-              }
-          ),
-          from!( // to query all courses that one user enrolled
-            "%agent_id",
-            link_type: "student->courses",
-            validation_package: || {
-                hdk::ValidationPackageDefinition::Entry
-            }              ,
-            validation: | _validation_data: hdk::LinkValidationData | {
-                // TODO: Homework. Implement validation rules if required.
-               Ok(())
-            }
-        ),
-        to!( // to query all enrolled user for a course
-            "%agent_id",
-            link_type: "course->students",
-            validation_package: || {
-                hdk::ValidationPackageDefinition::Entry
-            },
-            validation: | _validation_data: hdk::LinkValidationData | {
-                // TODO: Homework. Implement validation rules if required.
-                Ok(())
-            }
-        )
+            to!( // to query this course's latest data
+                "%agent_id", // QUESTION: what should I specify here?
```

Comment:
I don't understand if the `"%agend_id"` is another plain string or is this parameter specified according to some strict format. And looking at [the doc](https://docs.rs/hdk/0.0.41-alpha4/hdk/macro.to.html) doesn't help me understand how should I fill this parameter for the CourseAnchor.
@guillemcordoba could you please help me with that?

Reply:
So the `%agent_id` refers to the entry type that you're linking to/from. In normal circumstances, you can only put entry types you have defined somewhere in your application (in our case, the entry types we have are things like "course_data" and "module" and "content"). 

But there is an exception: when you want to link directly from/to an agent public address. In that case, you put the reserved "system type" `%agent_id`. 

In our case, we need to put "course_data" there.

Reply:
@guillemcordoba thank you for the clarification!